### PR TITLE
DE46838: alert when no quizzes

### PR DESCRIPTION
**Pull request by johngwilkinson**
Background: the server will, with the next release (new aq/fq API) return only quizzes with responses - see https://github.com/Brightspace/bdp-query/pull/170 - so it will be fairly common for there to be no quizzes appearing. This adds an alert about this.

Quality Notes
- functional
  - test w. demo page: empty data, and data with one result filtered out by a card filter
- security/performance/devops: n/a
- ux/docs
  - wording approved by PM

image

FYI @Brightspace/draco-vis

### CODE REVIEW

File: locales/en.js [hunk 1]
```diff
@@ -55,6 +55,8 @@ export default {
 	"dashboard:quizTruncationSubtext": "Please refine your filters or contact D2L Support for assistance.",
 	"dashboard:flaggedQuestionHeading": "Flagged Quiz Items",
 	"dashboard:flaggedQuestionMessage": "Quizzes have one or more flagged items for review.",
+	"dashboard:noResultsTitle": "There are no quizzes with attempts in the selected courses matching your filters.",
```
Comment (bemailloux) [inline locales/en.js:58]: Wow this feels like something of a run-on sentence

File: pages/assessment-quality-dashboard-view.js [hunk 2]
```diff
@@ -228,6 +228,10 @@ class AssessmentQualityDashboardView extends SortMixin(SkeletonMixin(Localizer(M
 		return this._filteredData.quizzes;
 	}
 
+	get _isDataEmpty() {
+		return !this.serverData.isLoading && this._itemsCount === 0;
```
Comment (bemailloux) [inline pages/assessment-quality-dashboard-view.js:232]: So the server will eventually filter out quizzes w/o attempts? That's not the current behaviour right? (Just wondering if I missed a PR)

**Review by bemailloux — APPROVED** [review]
LGTM!

File: pages/assessment-quality-dashboard-view.js (hunk 2 above)
Comment (bemailloux) [reply pages/assessment-quality-dashboard-view.js:232]: oh 🤦  
> see Brightspace/bdp-query#170

File: locales/en.js (hunk 1 above)
Comment (johngwilkinson) [reply locales/en.js]: Updated with slightly simpler language for the alert title sentence which the end-users can hopefully read better than what was there before which was indeed a bit of a run-on

**Review by NicholasHallman — APPROVED** [review]

File: pages/assessment-quality-dashboard-view.js [hunk 3]
```diff
@@ -367,6 +371,13 @@ class AssessmentQualityDashboardView extends SortMixin(SkeletonMixin(Localizer(M
 			</div>
 		${this._renderFilters()}
 		<h2 class="d2l-heading-3">${this.localize('dashboard:summary')}</h2>
+		<d2l-alert
+			class="d2l-no-quizzes-alert"
+			?hidden=${!this._isDataEmpty}
+			subtext=${this.localize('dashboard:noResultsSubtext')}
+			type="warning">
+			${this.localize('dashboard:noResultsTitle')}
```
[the remark this answers is not in the document]
Comment (johngwilkinson) [reply pages/assessment-quality-dashboard-view.js:379]: FYI: https://d2l.slack.com/archives/C0PHG3QB0/p1643141433156100

I expect to add `announce` in another PR, which will also replace any usages of `aria-live` (just one that I know of).

**Pull request merged**